libphobos/m4/druntime/libraries.m4

```diff
@@ -16,6 +40,17 @@ AC_ARG_ENABLE(thread-lib,
 ])
 
 
+# DRUNTIME_LIBRARIES_TIME
+# -----------------------
++# Check whether -lrt is required for sched_yield, nanosleep, or clock_gettime.
+AC_DEFUN([DRUNTIME_LIBRARIES_TIME],
+[
+AC_SEARCH_LIBS([sched_yield], [rt])
+AC_SEARCH_LIBS([nanosleep], [rt])
+AC_SEARCH_LIBS([clock_gettime], [rt])
+])
+
```

Comment:
Actually, it looks as though `pragma(lib)` does not reach the codegen pass if coming from an imported module not being compiled.  So perhaps embedding it in a debug section is not so good afterall, especially for MinGW where it seems that every other windows module has a `pragma(lib)` declaration.

As for the dlsym library, there's no mention of `pragma(lib, "dl")` anywhere, so relying on pragmas in code won't help either.

We could generate a second file that documents library dependencies for a given compilation, and have these installed with sources.  There's the `Target::loadModule` hook that can take care of scan for these after importing.  But there are some modules where we need to generate these dependencies ourselves for the benefit of linking phobos statically.